pkg/policy/validate.go

```diff
@@ -88,6 +90,121 @@ func Validate(policyRaw []byte, client *dclient.Client, mock bool, openAPIContro
 	return nil
 }
 
+// doesMatchAndExcludeConflict checks if the resultant
+// of match and exclude block is not an empty set
+func doesMatchAndExcludeConflict(rule kyverno.Rule) bool {
+
+	if reflect.DeepEqual(rule.MatchResources, kyverno.MatchResources{}) {
```

Comment:
Currently, `resource` is [required](https://github.com/nirmata/kyverno/blob/44d50c1cde4a9fa684b006850bf039812eded74b/definitions/install.yaml#L47-L48) in the openAPI schema, is this check needed? I think in CLI it has the schema built-in, right?

Reply:
This check is repeated in functions called before this and its in required in schema so we dont even need to check it, however we should keep it since its easy to reuse this function.

For example while implementing the new feature `deny` https://github.com/nirmata/kyverno/issues/744 match may become an optional field etc. Keeping the function like this prevents us from needing to make changes to it in the future.